xarray/backends/plugins.py

```diff
@@ -26,8 +26,8 @@ def remove_duplicates(entrypoints):
         unique_entrypoints.append(matches[0])
         matches_len = len(matches)
         if matches_len > 1:
-            selected_module_name = matches[0].module_name
-            all_module_names = [e.module_name for e in matches]
+            selected_module_name = matches[0].name
+            all_module_names = [e.name for e in matches]
```

Comment:
I think you should look at the `value` attribute instead of `name`. See how an EntryPoint object looks like `EntryPoint(name='rasterio', value='rioxarray.xarray_plugin:RasterioBackend', group='xarray.backends')`

```suggestion
            selected_module_name = matches[0].value
            all_module_names = [e.value for e in matches]
```

Reply:
Also it should be called `selected_cls_name` for clarity.